src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -862,16 +822,16 @@ private void SetupMultipleBuffers()
                         _multipleBufferGCHandles[i] = GCHandle.Alloc(_bufferListInternal[i].Array, GCHandleType.Pinned);
                     }
 
-                    if (_wsaBufferArray == null || _wsaBufferArray.Length < bufferCount)
+                    if (_wsaBufferArrayPinned == null || _wsaBufferArrayPinned.Length < bufferCount)
                     {
-                        _wsaBufferArray = new WSABuffer[bufferCount];
+                        _wsaBufferArrayPinned = GC.AllocateArray<WSABuffer>(bufferCount, pinned: true);
```

Comment:
Excuse me if I'm out of context, this might be silly. This code doesn't seem to read anything from the array. Can you allocate unitialized pinned array here or it's still not supported?

Reply:
It is possible to use AllocateUninitializedArray with “pinned: true”.
I was not sure if any code here relies on buffers being zero-initialized, so I did not do that, yet.